documentation/solution-architecture.md

```diff
@@ -0,0 +1,135 @@
+
+By using NLX certificates for signing and information retrieval, the waardepapieren service will align with REGULATION (EU) No 910/2014 for a large extend, the rest of which is to be deferred to the ongoing development of Discipl like as with ETSI TS 103 171.
+
+The Waardepapieren solution should adhere to EN 301 549 / WCAG 2.0 though this is largely dependent on the integration of the solution in the user interfaces at the side of the issuer and not the solution itself. The project does involve the implementation of a reference validator app  / wallet however which must adhere to EN 301 549 / WCAG 2.0.
+
+As said, the solution also aims at a W3C verifiable credential compliant solution. This compliancy is deferred to the ongoing development of Discipl however. The solution adheres to the main mandatory concepts of this and related standards however.
+
+## Organisational layer
+
+The following stakeholders are identified:
+
+- issuer : entity that issues verifiable claims. Also refers to an automated (abundance type of) service available at location of the issuer and/or through its website to act on behalf of the issuer.
+- clerk : employee that helps a client at location of issuer, services, identifies and authorizes the client manually
+- authorization service : automated service that identifies and authorizes client when client self services him/herself and accesses issuer through website
+- client : person in need of verifiable information required by a validator. It also becomes a holder of the resulting signed information.
+- validator : actor that needs to receive / verify information from client
+- administrator : actor that controls and monitors the automated issuer service
+
+An example of the essential conversation between real people as actors that would take place
+in the use cases the waardepapieren solution facilitates is given below:
+
+![](images/vcseq0.png)
+
+In essence this is the very process that also takes place when retrieving a proof through the waardepapieren solution as can be seen below in the other [verifiable claim sequence diagrams](images/vcseq.png).
+
+## Information layer
+
+All information is temporarily held in verifiable claim channels on the so called discipl ephemeral platform (http://github.com/discipl/discipl-core-ephemeral) in memory. This platform will enforce access rights denoted by the channel owner in specific allow claims within the channel, so by default all channels are only accessible to the channel owner holding a corresponding private key created and temporarily held by the channel owner. Of course the party hosting the ephemeral server and it's administrator need to be trusted also, but as this is the issuer as provider of the information it is by definition a party already trusted with this information by the client.
+
+The following channels will be created using this solution holding the following information (verifiable claims) with what kind of accessibility / retention:
+
```

Comment:
Do not understand what channels are, and how they are used. There is no explanation at the disciple-core link either.

Reply:
Ok, will work on this

Reply:
I added the diagram explaining the type of diagrams used in this document which contains a definition of the concept "channel" . I understand this all might still be too technical. I tried to make it visible how it really resembles the first diagram that displays a chat between actors which I hope are not perceived as technical. The difference is that actors are communicating through linked data triples instead of full sentences of natural language. In this document (which actually aims at a technical architecture too) I don want to simplify it further. Generic documentation around discipl core should address a more general functional explanation....

For your understanding: a channel is formed out of the linked data triples added to a discipl core supported (storage) platform and signed using the same public-private keypair... an actor might have more of those and thus communicate over multiple channels. A storage platform can optionally implement access control (who can read them) and data retention control (how long information is kept) on channels. There is always only 1 key with which data can be added to a channel and normally this key would be held by a single actor. So channels are groups of claims that will form naturally by doing communication using discipl core, and there's not much to manage around them. Does such an explanation help? Enough for now?

Reply:
To keep velocity I decide to merge this PR and improve documentation in additional PR's where needed